src/base/settingsstorage.h

```diff
@@ -45,7 +45,20 @@ struct IsQFlags : std::false_type {};
 template <typename T>
 struct IsQFlags<QFlags<T>> : std::true_type {};
 
-class SettingsStorage : public QObject
+class IHumanEditableSetting
+{
+public:
+    virtual ~IHumanEditableSetting() = default;
+
+    // requirement: T(const QString &) constructor for derived class `T`  // TODO: rewrite it in C++20 concept
```

Comment:
Yes, this is a problem. I wonder at what point in time would you plan to switch to c++20?

Reply:
Wasn't it discussed before? after we dropped Qt5 support.

Reply:
> Wasn't it discussed before? after we dropped Qt5 support.

This is something indefinite. That's why I didn't remember it.